web/src/settings_org.js

```diff
@@ -262,36 +262,68 @@ function set_msg_move_limit_setting(property_name) {
     enable_or_disable_related_message_move_time_limit_setting(property_name, disable_setting);
 }
 
-function message_delete_limit_setting_enabled(setting_value) {
+function message_delete_limit_setting_enabled() {
     // This function is used to check whether the time-limit setting
-    // should be enabled. The setting is disabled when delete_own_message_policy
-    // is set to 'admins only' as admins can delete messages irrespective of
-    // time limit.
-    if (setting_value === settings_config.common_message_policy_values.by_admins_only.code) {
+    // should be enabled. The setting is disabled when every user
+    // who is allowed to delete their own messages is also allowed
+    // to delete any message in the organization.
+    const realm_delete_own_message_policy = Number.parseInt(
+        $("#id_realm_delete_own_message_policy").val(),
+        10,
+    );
+    const realm_can_delete_any_message_group_id =
+        settings_components.get_dropdown_list_widget_setting_value(
+            $("#id_realm_can_delete_any_message_group"),
+        );
+    const realm_can_delete_any_message_group_name = user_groups.get_user_group_from_id(
+        realm_can_delete_any_message_group_id,
+    ).name;
+    const common_message_policy_values = settings_config.common_message_policy_values;
+
+    if (realm_delete_own_message_policy === common_message_policy_values.by_admins_only.code) {
+        return false;
+    }
+    if (realm_can_delete_any_message_group_name === "role:administrators") {
+        return true;
+    }
```

Comment:
I'll still say, I've read this code patiently and can confidently tell what it does now, and I'm sure its correct. However, there should be a conciser way of phrasing this.

First, find the lowest level of user that can delete own messages. Say,

```
guests = 0
members = 1
fullmembers = 2
moderators = 3
admins = 4
```

Then, let's say we compute `lowest_can_delete_own_messages = 2`

Similarly, `lowest_can_delete_any_message = 4`

Finally, 

```js
if (lowest_can_delete_own_messages < lowest_can_delete_any_message) {
    return true;
}
return false;
```

Reply:
The approach I used is kinda similar to what we do in `settings_data.user_has_permission`. Also, calculating a common value for both settings will still make the code long or am I missing something?

Reply:
I guess it boils down to personal preference, but no, the code shouldn't be big, or at least hard to parse.

```js
// random common_policy_values object
by_admins_only: {
        order: 1,
        code: 2,
        description: $t({defaultMessage: "Admins"}),
    },
```    

So, we want the `order` of the `code` we have in `realm_delete_own_message_policy`.

Something like:

```js
lowest_can_delete_own_messages = common_message_policy_values.find(e => {
    e.code === realm_delete_own_message_policy
  }).order;
// this sample doesn't error check if we couldn't find a match in the array yet.
```

I'm not sure where the "roll:administrators" strings are defined but presumably some "order"-like key also exists there?

Reply:
Actually `order` doesn't specify any priority in permission. It is just used to specify the order in which the values are shown in the dropdown in settings. Also, `role:...` are defined in `settings_config.system_user_groups_list` and there is no `order` key described there.

Reply:
In that case, let's just leave it as it is.

I still would rather add a "priority" key to `system_user_groups_list` and maybe add a `priority` key in addition to `order` key in that. Currently, the `order` key perfectly matches the increasing number of encompassing groups, but as you say that's only meant for the display order.

But we could make a note of this and if someone else later has to add another kind of user group, say, role:community-volunteers or whatever into this hierarchy, maybe they could make such changes later on.